aim/sdk/run.py

```diff
@@ -358,6 +361,11 @@ def _track_impl(
         if context is None:
             context = {}
 
+        if not is_number(value) and not isinstance(
+            value, (NoneType, bool, int, float, str, bytes, list, tuple, dict, Image)
+        ):
+            raise ValueError(f"Input metric of type {type(value)} is neither python number nor AimObject")
+
         try:
```

Comment:
lets split the type check into 2 parts:
1. if value is a custom object, no need to try to convert to number.
2. if `is_number(value)` returns True, try to convert to number, without catching exceptions.